src/bin/pg_autoctl/keeper.c

```diff
@@ -245,7 +245,7 @@ ReportPgIsRunning(Keeper *keeper)
 	int timeout = config->postgresql_restart_failure_timeout;
 	uint64_t now = time(NULL);
 
-	if (keeperState->current_role != PRIMARY_STATE)
+	if (keeperState->current_role != PRIMARY_STATE && keeperState->current_role != SINGLE_STATE)
```

Comment:
We should add a comment here to explain why we are interested in only those two states.

The monitor only cares about the PgIsRunning information when it deals with a PRIMARY, CATCHINGUP, and SECONDARY. Then we add in SINGLE state because we want to update our local values for that state and shut down the keeper when PostgreSQL is not running (for the 3rd time) when `--exit-...` has been used.

Maybe it would be even better to have this state depend on `keeper->config.exit_keeper_if_postgres_not_running` to make it obvious, and to avoid extra computations when they are not needed?